presto-hive/src/main/java/io/prestosql/plugin/hive/HivePageSourceProvider.java

```diff
@@ -119,9 +164,21 @@ public ConnectorPageSource createPageSource(ConnectorTransactionHandle transacti
                 hiveSplit.getTableToPartitionMapping(),
                 hiveSplit.getBucketConversion(),
                 hiveSplit.isS3SelectPushdownEnabled(),
-                hiveSplit.getAcidInfo());
-        if (pageSource.isPresent()) {
-            return pageSource.get();
+                hiveSplit.getAcidInfo(),
+                hiveTable.getTransaction());
+        if (optionalPageSource.isPresent()) {
+            ConnectorPageSource pageSource = optionalPageSource.get();
+            if (hiveTable.isDelete()) {
+                checkArgument(orcFileWriterFactory.isPresent(), "orcFileWriterFactory not supplied but required for DELETEs");
+                HivePageSource hivePageSource = (HivePageSource) pageSource;
+                OrcPageSource orcPageSource = (OrcPageSource) hivePageSource.getDelegate();
+                ColumnMetadata<OrcType> columnMetadata = orcPageSource.getColumnTypes();
+                HiveType rowType = fromOrcTypeToHiveType(columnMetadata.get(new OrcColumnId(6)), columnMetadata);
```

Comment:
~Do we have to compute `rowType` here? It is the same for all the tables right, so we should be just able to declare as a constant in `HiveUpdatablePageSource`.~

edit:

Ok I think I get it more. This is the type representing original table schema. Why do we need this one? I see that you are writing `null` value for it in `HiveUpdatablePageSource.deleteRows`. Does hive require it?
Could you maybe add a short comment in `HiveUpdatablePageSource.deleteRows` what is happening there.

Reply:
I asked that very same question of @electrum, saying "Gee, all I want to do is insert a null.  Why do I need a type?"
  The answer is that Apache ORC code we don't control insists that the layout of this "row" agree with the layout of an inserted row.  I added a comment to that effect.